### PR TITLE
feat(Image): Add configurator

### PR DESCRIPTION
**Pull request by soulcramer**
## 📋 Changes

Add configurator with the shape graphic lib to demonstrate how to clip to custom shapes

## 🤔 Context

Close [SPA-65](https://jira.ets.mpi-internal.com/browse/SPA-65)

## ✅ Checklist

- [x] I have reviewed the submitted code.
- [x] I have tested on a phone device/emulator.

## 📸 Screenshots

image

### CODE REVIEW

File: catalog/src/main/kotlin/com/adevinta/spark/catalog/configurator/samples/image/ImageConfigurator.kt [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+@Composable
+private fun ColumnScope.ImageSample() {
+    var state by remember { mutableStateOf(ImageState.Loading) }
+    var width by remember { mutableStateOf<Int?>(1) }
+    var height by remember { mutableStateOf<Int?>(1) }
+    var aspectRatio by remember { mutableStateOf(ImageAspectRatio.Custom) }
+    var imageShape by remember { mutableStateOf(ImageShape.Wavy) }
+    val painter =
+        rememberDrawablePainter(getDrawable(LocalContext.current, R.drawable.notifications))
+    val drawable = getDrawable(LocalContext.current, R.drawable.notifications)!!
```
Comment (SimonMarquis) [inline catalog/src/main/kotlin/com/adevinta/spark/catalog/configurator/samples/image/ImageConfigurator.kt:93]: Do we need to query getDrawable() twice?

File: spark/src/main/kotlin/com/adevinta/spark/components/textfields/SparkTextField.kt [hunk 2]
```diff
@@ -230,7 +230,6 @@ internal fun SparkTextField(
                 modifier
             }
                 .defaultMinSize(
-                    minWidth = TextFieldDefaults.MinWidth,
```
Comment (SimonMarquis) [inline spark/src/main/kotlin/com/adevinta/spark/components/textfields/SparkTextField.kt:233]: Unrelated change to be removed

File: spark/src/main/kotlin/com/adevinta/spark/components/image/Image.kt [hunk 3]
```diff
@@ -66,7 +66,7 @@ import com.adevinta.spark.tools.modifiers.sparkUsageOverlay
 
 @InternalSparkApi
 @Composable
-internal fun SparkImage(
+public fun SparkImage(
```
Comment (SimonMarquis) [inline spark/src/main/kotlin/com/adevinta/spark/components/image/Image.kt:69]: Can we keep it internal and add a flag to use it in the configurator?

Comment (soulcramer) [reply spark/src/main/kotlin/com/adevinta/spark/components/image/Image.kt:69]: Oh I was wandering how we can do it without triggering a compiler warning 🤔 
Do you have another solution than https://publicobject.com/2024/01/30/internal-visibility/ ?

I tried to use it but unfortunatly the compiler give me this warning

> This code uses error suppression for 'INVISIBLE_REFERENCE'. While it might compile and work, the compiler behavior is UNSPECIFIED and WON'T BE PRESERVED. Please report your use case to the Kotlin issue tracker instead: https://kotl.in/issue

Comment (SimonMarquis) [reply spark/src/main/kotlin/com/adevinta/spark/components/image/Image.kt:69]: This was the solution I was going to give you. There might be additional compiler options to turn this warning off.
Otherwise, you can try the Annotation that restrict usage to a specific library (I think AndroidX uses this a lot, but I don't remember the name of this annotation).

Comment (soulcramer) [reply spark/src/main/kotlin/com/adevinta/spark/components/image/Image.kt:71]: Should be resolved in e46e6116208ab87c0a0372dab5223b1689751000
